File: frontend/src/api/people.ts
import client from './client'

// ============ Types ============

export interface PersonPlatform {
  id: number
  platform: string
  handle: string
  profile_url: string
  follower_count: number
  is_verified: boolean
  scrape_enabled: boolean
  last_checked: string | null
}

export interface Person {
  id: number
  name: string
  type: string
  tier: string | null
  bio: string | null
  primary_region: string | null
  secondary_regions: string[] | null
  demographics: string[] | null
  style_tags: string[] | null
  categories: string[] | null
  follower_count_total: number
  relevance_score: number
  active: boolean
  scrape_frequency: string
  priority: number
  notes: string | null
  last_scraped_at: string | null
  added_at: string
  platforms: PersonPlatform[]
}

export interface PersonStats {
  total_people: number
  by_type: Record<string, number>
  by_region: Record<string, number>
  by_tier: Record<string, number>
  total_platforms: number
  active_count: number
}

export interface ScrapedPost {
  id: number
  person_id: number
  platform: string
  post_url: string
  image_urls: string[]
  caption: string | null
  hashtags: string[]
  likes: number
  comments: number
  shares: number
  views: number
  analyzed: boolean
  ai_analysis: Record<string, any> | null
  scraped_at: string
  posted_at: string | null
}

// ============ API Functions ============

export const getPeopleStats = async (): Promise<PersonStats> => {
  const response = await client.get<PersonStats>('/people/stats')
  return response.data
}

export const listPeople = async (params?: {
  type?: string
  tier?: string
  region?: string
  platform?: string
  active?: boolean
  search?: string
  sort_by?: string
  limit?: number
  offset?: number
}): Promise<Person[]> => {
  const response = await client.get<Person[]>('/people', { params })
  return response.data
}

export const getPerson = async (id: number): Promise<Person> => {
  const response = await client.get<Person>(`/people/${id}`)
  return response.data
}

export const addPerson = async (data: {
  name: string
  type: string
  tier?: string
  bio?: string
  primary_region?: string
  platforms: Array<{
    platform: string
    handle: string
    follower_count?: number
  }>
}): Promise<Person> => {
  const response = await client.post<Person>('/people', data)
  return response.data
}

export const updatePerson = async (
  id: number,
  data: Partial<{
    name: string
    type: string
    tier: string
    bio: string
    primary_region: string
    active: boolean
    priority: number
    relevance_score: number
  }>
): Promise<Person> => {
  const response = await client.put<Person>(`/people/${id}`, data)
  return response.data
}

export const deletePerson = async (id: number): Promise<void> => {
  await client.delete(`/people/${id}`)
}

export const seedPeopleDatabase = async (): Promise<{
  message: string
  created: number
  skipped: number
  errors: Array<{ name: string; error: string }>
  total_in_seed: number
}> => {
  const response = await client.post('/people/seed')
  return response.data
}

export const scrapePerson = async (id: number): Promise<{
  status: string
  new_posts: number
  person: string
}> => {
  const response = await client.post(`/people/${id}/scrape`)
  return response.data
}

export const getPersonPosts = async (
  id: number,
  params?: { limit?: number; analyzed_only?: boolean }
): Promise<ScrapedPost[]> => {
  const response = await client.get<ScrapedPost[]>(`/people/${id}/posts`, { params })
  return response.data
}

export const scrapeBatch = async (params?: {
  type?: string
  region?: string
  priority_max?: number
  limit?: number
}): Promise<{
  total_people: number
  total_new_posts: number
  errors: Array<{ name: string; error: string }>
}> => {
  const response = await client.post('/people/scrape-batch', null, { params })
  return response.data
}

File: backend/app/schemas/people.py
"""Pydantic schemas for People database."""

from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


# --- Person ---

class PersonPlatformCreate(BaseModel):
    platform: str  # instagram, tiktok, twitter, pinterest, youtube
    handle: str
    profile_url: Optional[str] = None
    follower_count: int = 0
    is_verified: bool = False


class PersonCreate(BaseModel):
    name: str
    type: str  # celebrity, influencer, brand, editor, stylist, media
    tier: Optional[str] = None
    bio: Optional[str] = None
    primary_region: Optional[str] = None
    secondary_regions: Optional[List[str]] = None
    demographics: Optional[List[str]] = None
    style_tags: Optional[List[str]] = None
    categories: Optional[List[str]] = None
    scrape_frequency: str = "daily"
    priority: int = 5
    notes: Optional[str] = None
    platforms: List[PersonPlatformCreate] = []


class PersonBulkCreate(BaseModel):
    people: List[PersonCreate]


class PersonPlatformResponse(BaseModel):
    id: int
    person_id: int
    platform: str
    handle: str
    profile_url: Optional[str]
    follower_count: int
    engagement_rate: float
    is_verified: bool
    scrape_enabled: bool
    last_checked: Optional[datetime]

    class Config:
        from_attributes = True


class PersonResponse(BaseModel):
    id: int
    name: str
    type: str
    tier: Optional[str]
    bio: Optional[str]
    primary_region: Optional[str]
    secondary_regions: Optional[List[str]]
    demographics: Optional[List[str]]
    style_tags: Optional[List[str]]
    categories: Optional[List[str]]
    follower_count_total: int
    avg_engagement_rate: float
    relevance_score: float
    active: bool
    scrape_frequency: str
    priority: int
    added_at: datetime
    last_scraped_at: Optional[datetime]
    notes: Optional[str]
    platforms: List[PersonPlatformResponse] = []

    class Config:
        from_attributes = True


class PersonUpdate(BaseModel):
    name: Optional[str] = None
    type: Optional[str] = None
    tier: Optional[str] = None
    bio: Optional[str] = None
    primary_region: Optional[str] = None
    demographics: Optional[List[str]] = None
    style_tags: Optional[List[str]] = None
    categories: Optional[List[str]] = None
    active: Optional[bool] = None
    scrape_frequency: Optional[str] = None
    priority: Optional[int] = None
    relevance_score: Optional[float] = None
    notes: Optional[str] = None


class PersonStats(BaseModel):
    total_people: int
    by_type: dict
    by_region: dict
    by_tier: dict
    total_platforms: int
    active_count: int


# --- Scraped Posts ---

class ScrapedPostResponse(BaseModel):
    id: int
    person_id: int
    platform: str
    post_url: str
    image_urls: Optional[List[str]]
    caption: Optional[str]
    hashtags: Optional[List[str]]
    likes: int
    comments: int
    shares: int
    views: int
    analyzed: bool
    category: Optional[str]
    colors: Optional[List[str]]
    style_tags: Optional[List[str]]
    ai_narrative: Optional[str]
    posted_at: Optional[datetime]
    scraped_at: datetime

    class Config:
        from_attributes = True
